src/managers/balancer.py

```diff
@@ -18,6 +22,93 @@
 
 logger = logging.getLogger(__name__)
 
+RebalanceMode = Literal["full", "add", "remove"]
+
+
+class CruiseControlClient:
+    """Client wrapper for CruiseControl."""
+
+    def __init__(self, username: str, password: str, host: str = "localhost", port: int = 9090):
```

Comment:
**suggestion**: I would advise adding a parameter `client` or `session` like this `client: requests.Session | None = None`.

Then, you may instantiate its value in the init method if necessary. The two benefits are:
- you would define the auth only there. Same thing if we need to specify a proxy or a cert at some point.
- it will be convenient to pass our own custom session in tests, without mocking anything

Reply:
I've added something similar, but kept it as it is, does it look OK?